teps/0044-decouple-task-composition-from-scheduling.md

```diff
@@ -84,97 +84,399 @@ that can be shared between them, and running the second Task will be delayed by
+   possibly even the names of the related TaskRuns
+
+Pros:
+* This maintains the existing relationships between the Pipelines resources
+  (N Tasks -> 1 Pipeline -> 1 PipelineRun -> N TaskRun)
+* Having the Pipeline controller create TaskRuns up front (as "pending" or similar) might have other benefits, for
+  example we've struggled in the past with how to represent the status of Tasks in a Pipeline which don't have a
+  backing TaskRun, e.g. they are skipped or cancelled. Now there actually would be a TaskRun backing them.
+  * This would not be possible for any dynamically created TaskRuns, e.g. dynamic looping based on Task results
+    (from [TEP-0090 matrix support](https://github.com/tektoncd/community/pull/532))
+
+Cons:
+* Have to create all TaskRuns in advance (at least in this specific scenario)
+* Creating the underlying pod is more complicated than other options in that:
+  * It happens across multiple reconciles
+  * It requires state to be maintained across reconciles
+  * It requires co-ordination information about the Pipeline as a whole to be contained in corresponding TaskRuns
+
+#### Other implementation options
+
+##### Rely on the affinity assistant
+
+In this approach we'd rely on [the affinity assistant](https://github.com/tektoncd/community/pull/318/files) to
+co-locate pods that use the same workspace.
+
+Cons:
+* Doesn't help us with the overhead of multiple pods
+* [TEP-0046](https://github.com/tektoncd/community/pull/318/files) explores shortcomings of this approach
+
+##### Custom scheduler
+
+In this approach we use a custom scheduler to schedule our pods.
+
+Cons:
+* [@jlpetterson has explored this option](https://docs.google.com/document/d/1lIqFP1c3apFwCPEqO0Bq9j-XCDH5XRmq8EhYV_BPe9Y/edit#heading=h.18c0pv2k7d1a)
+  and felt it added more complexity without much gain (see also https://github.com/tektoncd/pipeline/issues/3052)
+  * For example the issue [#3049](https://github.com/tektoncd/pipeline/issues/3049)
+* Doesn't help us with the overhead of multiple pods (each Task would still be a pod)
+
+##### Support other ways to share data (e.g. buckets)
+
+In this approach we could add more workspace types that support other ways of sharing data between pods, for example
+uploading to and downloading from s3. (See [#290](https://github.com/tektoncd/community/pull/290/files).)
+
+[This is something we support for "linking" PipelineResources.](https://github.com/tektoncd/pipeline/blob/master/docs/install.md#configuring-pipelineresource-storage)
+
+Pros:
+* Easier out of the box support for other ways of sharing data
+
+Cons:
+* Uploading and downloading at the beginning and end of every Task is not as efficient as being able to share the same
+  disk
+* We'd need to define an extension mechanism so folks can use whatever backing store they want
+* Doesn't help us with the overhead of multiple pods (each Task would still be a pod)
+
```

Comment:
I think this list is missing something(s) like 
- "grouping" task proposal ? (using a field or something, or even a "Task" that is of type TaskGroup).
- pre/post task hooks (matt's TEP proposal)